bin/test_combos

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash -e
+#
+# Test combinations of traces with test site models
+#
+
+tests=tests/combos/*.json
+
+for test in $tests; do
+    echo '################' Running combo test $test
+    traces=$(jq -r .traces $test)
+    site_model=$(jq -r .site_model $test)
+    validator_config=$(jq -r .validator $test)
+    if [[ $validator_config != null ]]; then
+        validator/bin/validate -- schema trace $traces $site_model
+        check=$(echo $validator_config | jq -r .check)
+        echo $check
+        eval $check
```

Comment:
Bare/unprotected eval (related: using bare 'source', in PR 561) continues to be a red flag to me.

Reply:
> Bare/unprotected eval (related: using bare 'source', in PR 561) continues to be a red flag to me.

What would you suggest instead?  This is just implementing execution of the contents of tests/combos/missing_pointset.json -- which is trying to find the right balance of flexibility with structure.

Reply:
My simplest suggestion, i.e. least delta from what we have now, versus changing the test rig overall, just make it more explicit:

- in the test definition write something structured like "validator": {  { "fgrep_check": { "path":... , "pattern":...}}   and here when you perform the fgrep, quote the arguments

Reply:
This is more or less modeled after the way github actions does things --https://github.com/faucetsdn/udmi/blob/master/.github/workflows/testing.yml#L59 -- which just runs the command line as given. Seems to work reasonably well in the context...

Reply:
I don't see the line you're comparing to. The testing yaml is just running commands that were already written, right? My concern with eval is that if you can change the shell or invoke something else.  I realize we are invested in the shell script tests but we could at least be as careful as possible here. Adding a fgrep handler shouldn't be too hard?